client/src/modules/shopping-list/components/ArchivedList.jsx

```diff
@@ -0,0 +1,76 @@
+import React, { Fragment, PureComponent } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+
+import DialogBox from 'common/components/DialogBox';
+import { deleteList, restoreList } from 'modules/shopping-list/model/actions';
+
+class ArchivedList extends PureComponent {
+  state = {
+    showDialogBox: false
+  };
+
+  showDialogBox = () => {
+    this.setState({ showDialogBox: true });
+  };
+
+  hideDialogBox = () => {
+    this.setState({ showDialogBox: false });
+  };
+
+  restoreListHandler = listId => () => {
+    const { restoreList } = this.props;
+    restoreList(listId);
+  };
+
+  deleteListHandler = id => () => {
+    const { deleteList } = this.props;
```

Comment:
Because clicking on 'permanently delete' button shouldn't invoke delete action directly but display dialog to confirm this delete action... So you have to return function which holds a reference to list Id, and this returned function will be invoke when you click confirm button on the dialog

Reply:
![zrzut ekranu 2019-02-25 o 10 34 15](https://user-images.githubusercontent.com/27632432/53327837-3a68e880-38e9-11e9-883d-602e3c8ae341.png)
![zrzut ekranu 2019-02-25 o 10 34 24](https://user-images.githubusercontent.com/27632432/53327838-f2e25c80-38e8-11e9-94cc-2c0b5af3d63d.png)

Try this solution, it's more intuitive and common pattern for other developers to understand